WebSocket/hub.go

```diff
@@ -55,22 +56,35 @@ type hub struct {
 	logMx sync.RWMutex
 	log   [][]byte
 
-	organizer *actors.Organizer
-	witness   *actors.Witness
+	actor actors.Actor
 
 	connIndex int
 }
 
-func NewHub() *hub {
+func NewOrganizerHub(pkey string) *hub {
+	return newHub("o", pkey)
+}
+
+func NewWitnessHub(pkey string) *hub {
+	return newHub("w", pkey)
+}
+
+func newHub(mode string, pkey string) *hub {
 
 	h := &hub{
 		connectionsMx:   sync.RWMutex{},
 		receivedMessage: make(chan []byte),
 		connections:     make(map[*connection]struct{}),
 		connIndex:       0,
 		idOfSender:      -1,
-		organizer:       actors.NewOrganizer("", "orgDatabase.db"),
-		witness:         actors.NewWitness("", "witDatabase.db"),
+	}
+
+	if mode == "o" {
+		h.actor = actors.NewOrganizer(pkey, "orgdatabase.db")
+	} else if mode == "w" {
+		h.actor = actors.NewWitness(pkey, "witdatabase.db")
```

Comment:
yes, you're right, we should modify this.
Though this never overwrites anything, it's just going to open the files. This will cause both server to run slower (because of lock requests on the file when it is accessed), but as the protocol and our database is defined, there should be no data loss. The real problem is that a witness could access info "it should not" access, like channel he's not subscribed to for example